src/vivarium_gates_child_iv_iron/components/fertility.py

```diff
@@ -65,7 +65,8 @@ def on_time_step(self, event: Event) -> None:
         birth_records = self.birth_records
         born_previous_step_mask = (birth_records['birth_date'] < self.clock()) & (
             birth_records['birth_date'] > self.clock() - event.step_size)
-        born_previous_step = birth_records[born_previous_step_mask]
+        born_previous_step = birth_records[born_previous_step_mask].copy()
+        born_previous_step.loc[:, "maternal_id"] = born_previous_step.index
```

Comment:
Nit: Since most pandas operations are not in-place, you could do this instead of using `copy`:
```
born_previous_step = birth_records[born_previous_step_mask].reset_index().rename({"index": "maternal_id"})
```
No need to change this though.

Reply:
Clever!